compiler/semantic/sql.go

```diff
@@ -478,9 +478,40 @@ func (t *translator) sqlQueryBody(query ast.SQLQueryBody, seq sem.Seq) (sem.Seq,
 		return seq, sch
 	case *ast.SQLUnion:
 		left, leftSch := t.sqlQueryBody(query.Left, seq)
-		left, _ = endScope(query.Left.(ast.Node), leftSch, left)
+		left, leftSch = endScope(query.Left.(ast.Node), leftSch, left)
+		leftCols, lok := leftSch.outColumns()
+		if !lok {
+			t.error(query.Left, errors.New("set operations cannot be applied to dynamic sources"))
+		}
 		right, rightSch := t.sqlQueryBody(query.Right, seq)
-		right, _ = endScope(query.Right.(ast.Node), rightSch, right)
+		right, rightSch = endScope(query.Right.(ast.Node), rightSch, right)
+		rightCols, rok := rightSch.outColumns()
+		if !rok {
+			t.error(query.Right, errors.New("set operations cannot be applied to dynamic sources"))
+		}
```

Comment:
Add test coverage for this.

Reply:
I wasn't quite sure how to test for a dynamic source that won't have to be deleted soon. Like I could query against a jsup file but aren't we planning to read types of small jsup files?

Reply:
> I wasn't quite sure how to test for a dynamic source that won't have to be deleted soon. Like I could query against a jsup file but aren't we planning to read types of small jsup files?

Let's cross that bridge when we come to it.